static/src/js/zustand/selectors/preferences.js

```diff
@@ -0,0 +1,48 @@
+import { translateDefaultCollectionSortKey } from '../../util/collections'
+
+/**
+ * Get preferences from Zustand store
+ */
+export const getPreferences = (state) => state.preferences.preferences
+
+/**
+ * Get map preferences from Zustand store
+ */
+export const getMapPreferences = (state) => {
+  const preferences = getPreferences(state)
+
+  return preferences?.mapView
+}
+
+/**
+ * Get collection sort preference from Zustand store
+ */
+export const getCollectionSortPreference = (state) => {
+  const preferences = getPreferences(state)
+
+  return preferences?.collectionSort
+}
+
+/**
+ * Creates a selector function for collection sort key parameter with preference comparison
+ * Returns a function that takes paramCollectionSortKey and returns null if it matches user preference
+ */
+export const getCollectionSortKeyParameterSelector = (state) => {
+  const preferences = getPreferences(state)
+  const userPrefSortKey = preferences?.collectionSort
+  const translatedUserPrefSortKey = translateDefaultCollectionSortKey(userPrefSortKey)
+
+  // Return a function that can be used to compare against paramCollectionSortKey
```

Comment:
Selectors can't return a function, they return a value. This selector should take in the zustand state, fetch the `paramCollectionSortKey` from redux, and then do the same logic the previous redux selector did. You can fetch the value from redux using this code

```
const {
    getState: reduxGetState
  } = configureStore()
  const reduxState = reduxGetState() 
  const paramCollectionSortKey = getParamCollectionSortKey(reduxState) // this is the old redux selector
```